### PR TITLE
Add SetIfEmpty transformer

### PR DESCRIPTION
**Pull request by eorgadpalo**
## Status
Ready

## Description
Adding a transformer to set value to a preset default one if the value is evaluated to be None or empty + Unit tests

## Fixes:
https://github.com/demisto/etc/issues/18665

## Does it break backward compatibility?
   - No

## Technical writer review
- [ X] [YAML file](https://github.com/demisto/content/pull/4876/files#diff-444b799a77d8dfd806603ad1f1792ac7)

### CODE REVIEW

File: Scripts/SetIfEmpty/SetIfEmpty.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+
+def main(args):
+    demisto.results(getValueToSet(args))
+
+def getValueToSet(args):
+    value = args.get('value')
+    applyIfEmpty = True if args.get('applyIfEmpty') == 'true' else False
```
Comment (asafshen) [inline Scripts/SetIfEmpty/SetIfEmpty.py]: lets support a case when user write `True`/`TRUE` as well

```suggestion
    applyIfEmpty = True if args.get('applyIfEmpty', '').lower() == 'true' else False
```

File: Scripts/SetIfEmpty/SetIfEmpty.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+args:
+- default: false
+  description: The string or dictionary value to check if empty.
```
Comment (asafshen) [inline Scripts/SetIfEmpty/SetIfEmpty.yml]: not necessarily string or dictionary

File: Scripts/SetIfEmpty/SetIfEmpty.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  description: If set to true, empty strings and dictionaries will be identified as
+    invalid values and the default value will be returned. True by default.
+  isArray: false
+  name: applyIfEmpty
+  predefined:
+  - 'true'
+  - 'false'
+  required: false
+  secret: false
+comment: Todo
```
Comment (asafshen) [inline Scripts/SetIfEmpty/SetIfEmpty.yml]: need to have a descriptive comment

File: Scripts/SetIfEmpty/SetIfEmpty.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+commonfields:
+  id: 6debcb35-368b-4003-9d1e-445a16553735
```
Comment (asafshen) [inline Scripts/SetIfEmpty/SetIfEmpty.yml]: need to set id by the convention (I think its the same as the name)

File: Scripts/SetIfEmpty/SetIfEmpty.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  name: defaultValue
+  predefined:
+  - ''
+  required: true
+  secret: false
+- auto: PREDEFINED
+  default: true
+  defaultValue: 'true'
+  description: If set to true, empty strings and dictionaries will be identified as
+    invalid values and the default value will be returned. True by default.
```
Comment (asafshen) [inline Scripts/SetIfEmpty/SetIfEmpty.yml]: - `be identified as invalid values` - the values are not invalid, maybe something like `be handled as empty value
- no need to write `default is true`

File: Scripts/SetIfEmpty/SetIfEmpty_test.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
+# If applyIfEmpty default is changed to false the script will bug BUT THE TESTS WON'T BREAK
```
Comment (asafshen) [inline Scripts/SetIfEmpty/SetIfEmpty_test.py]: lets remove this comment

**Comment by AmirSimon** [comment]
@eorgadpalo -
image

address also the errors of circleCi, commit and push the fix.

**Comment by Itay4** [comment]
is there an issue for this?

File: Scripts/SetIfEmpty/SetIfEmpty.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+
+def main(args):
+    demisto.results(getValueToSet(args))
+
+def getValueToSet(args):
```
Comment (Itay4) [inline Scripts/SetIfEmpty/SetIfEmpty.py]:
```suggestion
def get_value_to_set(args):
```

File: Scripts/SetIfEmpty/SetIfEmpty.py [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    value = args.get('value')
+    applyIfEmpty = True if args.get('applyIfEmpty', '').lower() == 'true' else False
```
Comment (Itay4) [inline Scripts/SetIfEmpty/SetIfEmpty.py]:
```suggestion
    apply_if_empty = True if args.get('applyIfEmpty', '').lower() == 'true' else False
```

File: Scripts/SetIfEmpty/SetIfEmpty.yml [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  isArray: false
+  name: value
+  required: false
+  secret: false
+- default: false
+  description: The new value to return if the original value was empty.
+  isArray: false
+  name: defaultValue
+  predefined:
+  - ''
```
Comment (Itay4) [inline Scripts/SetIfEmpty/SetIfEmpty.yml]: remove empty arg

File: Scripts/SetIfEmpty/SetIfEmpty_test.py [hunk 10]
```diff
@@ -0,0 +1,52 @@
+
+from SetIfEmpty import getValueToSet
+
+def test_WhenValueIsAValidString_ShouldReturnValue():
```
Comment (Itay4) [inline Scripts/SetIfEmpty/SetIfEmpty_test.py]: please change to pep8 convention

File: Scripts/SetIfEmpty/SetIfEmpty.py (hunk 7 above)
Comment (Itay4) [inline Scripts/SetIfEmpty/SetIfEmpty.py]: this function should be placed the `main` func

Comment (eorgadpalo) [reply Scripts/SetIfEmpty/SetIfEmpty.py]: Did you mean above the main function?

Comment (Itay4) [reply Scripts/SetIfEmpty/SetIfEmpty.py]: yes

File: Scripts/SetIfEmpty/SetIfEmpty_test.py (hunk 10 above)
Comment (eorgadpalo) [reply Scripts/SetIfEmpty/SetIfEmpty_test.py]: Regarding the test names? 
Couldn't find anything regarding unit test naming in this guide:
https://www.python.org/dev/peps/pep-0008/

Looking at other files, did you mean naming the tests (for example) 
test_WhenValueIsEmptyDictionary_ShouldReturnDefaultValue  -->
test_value_is_empty_dictionary ?

File: Scripts/SetIfEmpty/SetIfEmpty.yml [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+commonfields:
+  id: SetIfEmpty
+  version: -1
+enabled: false
+name: SetIfEmpty
+script: '-'
+subtype: python3
+system: false
+tags:
+- transformer
```
Comment (asafshen) [inline Scripts/SetIfEmpty/SetIfEmpty.yml]: it must have an `entirelist` tag, otherwise it doesn't work properly

**Comment by richardbluestone** [comment]
Reviewed the yml file

**Comment by idovandijk** [comment]
@eorgadpalo just mentioning that in order to make sure the issue you worked on is closed, you should specify "fixes: [issue link]" as opposed to having a "fixes" title in this PR :)
closing the other issue since I see that you did it

**Pull request merged**